caluma/caluma_core/tests/test_health_checks.py

```diff
@@ -0,0 +1,445 @@
+    """Mock minio to produce exception when calling get_object or stat_object.
+
+    This scenario might occur when there is a internal minio issue or if a
+    previous put_request isn't successful (which doesn't issue exception).
+    """
+    # raise exception instead of handling request
+    def side_effect(*_):
+        raise Exception()
+
+    # patch minio get and put request
+    mocker.patch.object(Minio, "get_object")
+    mocker.patch.object(Minio, "put_object")
+
+    # replace return values
+    Minio.stat_object.side_effect = side_effect
+    Minio.get_object.side_effect = side_effect
+    Minio.put_object.return_value = None
+
+
+def test_minio_read_error(client, snapshot, db, minio_mock_read_error):
+    """Test /healthz/ endpoint response when minio receives a read error.
+
+    Mocks minio read functions to produce an error.
+    Affects minio health check get_object and stat_object functions.
+    """
+    # get /healthz/ response and compare to previous snapshot
+    response = client.get(reverse("healthz"))
+    snapshot.assert_match(response.json())
+
+    # assert minio check fails
+    assert response.json()["minio"] == {"ok": False}
+    assert response.status_code == watchman_settings.WATCHMAN_ERROR_CODE
+
+
+def test_db_migrations_applied(client, snapshot, db, minio_mock_working):
+    """Test /healthz/ endpoint response when migrations all created & applied.
+
+    Ensure no changes or unapplied migrations are detected.
+    DB migrations health check should pass.
+    """
+    # manually migrate db to ensure the database is in the newest state and
+    # no unapplied migrations are still around
+    # when running migration tests database might be left in an
+    # inconsistent state
+    management.call_command("migrate")
+
+    # get /healthz/ response and compare to previous snapshot
+    response = client.get(reverse("healthz"))
+    snapshot.assert_match(response.json())
+
+    # assert db migrations check passes
+    assert response.json()["db migrations"] == [{"default": {"ok": True}}]
+    assert response.status_code == 200
+
+
+def test_db_migrations_not_applied(client, snapshot, db, minio_mock_working):
+    """Test /healthz/ endpoint response when migrations haven't been applied.
+
+    Undo applied migrations for app 'contenttypes'.
+    Affects db migrations health check, which should detect unapplied
+    migrations and fail.
+    """
+    # undo applied migrations for app contenttypes
+    management.call_command("migrate", "contenttypes", "zero")
+
+    # get /healthz/ response and compare to previous snapshot
+    response = client.get(reverse("healthz"))
+    snapshot.assert_match(response.json())
+
+    # assert db migrations check fails
+    assert response.json()["db migrations"] == [{"default": {"ok": False}}]
+    assert response.status_code == watchman_settings.WATCHMAN_ERROR_CODE
+
+    # re-apply migrations for app contenttypes
+    management.call_command("migrate", "contenttypes")
```

Comment:
Is this really needed? If yes, I'd put it in a fixture, to make sure it's also run, in case of an exception:

```python
@pytest.fixture
def roll_back_migrations():  # maybe you come up with a better name ;)
    # undo applied migrations for app contenttypes
    management.call_command("migrate", "contenttypes", "zero")

    yield

    # re-apply migrations for app contenttypes
    management.call_command("migrate", "contenttypes")
```